### PR TITLE
ASIM Authentication schema parser with its sample and test data for SentinelOne

### PR DESCRIPTION
**Pull request by jayeshprajapaticrest**
Required items, please complete
   
   Change(s):
   - Added ASIM Authentication schema parser for SentinelOne based on API Data

   Reason for Change(s):
   - Initial version ASIM Authentication Parser

   Version Updated:
   - Initial version

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by v-rusraut** [comment]
Hi @jayeshprajapaticrest,
Thank you for raising Pull Request with us! We will review it internally and get back to you by shortly.

**Comment by vakohl** [comment]
@jayeshprajapaticrest Will try to review this within this week

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      activityType_d: real,
+      EventType: string,
+      EventResult: string,
+      EventOriginalType: string
+  )
+  [
+      27, "Logon", "Success", "User Logged In",
+      33, "Logoff", "Success", "User Logged Out",
+      133, "Logon", "Failure", "Existing User Login Failure",
+      134, "Logon", "Failure", "Unknown User Login",
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:44]: "Unknown User Login" is Failed login or Success?

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+      "kubernetes node", "Other",
+      "unknown", "Other"
+  ];
+  let parser = (disabled: bool=false) {
+      let alldata = SentinelOne_CL
+          | where not(disabled);
+      let activitydata = alldata
+          | where event_name_s == "Activities."
+              and activityType_d in (27, 33, 133, 134, 139, 3629)
+          | parse-kv DataFields_s as (ipAddress: string, username: string, userScope: string, accountName: string, fullScopeDetails: string, fullScopeDetailsPath: string, role: string, scopeLevel: string, source: string, sourceType: string) with (pair_delimiter=",", kv_delimiter=":", quote='"')
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:82]: 'role' could potentially be TargetUserType?
https://learn.microsoft.com/en-us/azure/sentinel/normalization-about-schemas#additional-user-fields

File: Parsers/ASimAuthentication/Parsers/vimAuthenticationSentinelOne.yaml [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+      "INTERACTIVE", "Interactive",
+      "NETWORK_CLEAR_TEXT", "Remote",
+      "NETWORK_CREDENTIALS", "Remote",
+      "NETWORK", "Remote",
+      "REMOTE_INTERACTIVE", "RemoteInteractive",
+      "SERVICE", "Service",
+      "SYSTEM", "System",
+      "UNLOCK", "System"
+  ];
+  let DeviceTypeLookup = datatable (agentDetectionInfo_machineType_s: string, TargetDeviceType: string)
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/vimAuthenticationSentinelOne.yaml:77]: SrcDeviceType?

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+          EventCount = int(1),
+          EventProduct = "SentinelOne",
+          EventSchemaVersion = "0.1.3",
+          EventVendor = "SentinelOne",
+          EventSchema = "Authentication"
+      | extend
+          Dvc = coalesce(DvcHostname, DvcIpAddr, EventProduct),
+          SrcIpAddr = DvcIpAddr,
+          EventEndTime = EventStartTime,
+          EventUid = _ResourceId,
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:142]: pls map with _ItemId
https://learn.microsoft.com/en-us/azure/sentinel/normalization-common-fields#eventuid

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+      134, "Logon", "Failure", "Unknown User Login",
+      139, "Logon", "Failure", "User Failed to Start an Unrestricted Session",
+      3629, "Logon", "Success", "Login Using Saved 2FA Recovery Code"
+  ];
+  let EventTypeLookup = datatable (alertInfo_eventType_s: string, EventType: string)
+      [
+      "WINLOGONATTEMPT", "Logon",
+      "WINLOGOFFATTEMPT", "Logoff"
+  ];
+  let EventSubTypeLookup = datatable (alertInfo_loginType_s: string, EventSubType: string)
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:53]: pls map original value with EventOriginalSubType

File: Parsers/ASimAuthentication/Parsers/vimAuthenticationSentinelOne.yaml [hunk 6]
```diff
@@ -0,0 +1,194 @@
+Parser:
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/vimAuthenticationSentinelOne.yaml:1]: same comments applicable for vim as provided in asim parser

**Comment by vakohl** [comment]
@jayeshprajapaticrest pls resolve validation error

**Review by vakohl — CHANGES_REQUESTED** [review]

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 7]
```diff
@@ -0,0 +1,182 @@
+Parser:
+  Title: ASIM Authentication parser for SentinelOne
+  Version: '0.1'
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:3]: pls use 0.1.0

**Comment by v-sudkharat** [comment]
Hi @jayeshprajapaticrest, hope you are doing well. Just wanted to check if you got a chance to look at the suggestions shared. Please feel free to reach out to us for any queries or support. Thanks!

**Comment by v-atulyadav** [comment]
Hi @jayeshprajapaticrest,
We are still waiting for your response to the above comments. Thanks

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml (hunk 1 above)
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:44]: As discussed, "Unknown User Login" template indicates that Unknown user tried to login but failed so we kept EventResult as Failure.

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+      let alldata = SentinelOne_CL
+          | where not(disabled);
+      let activitydata = alldata
+          | where event_name_s == "Activities."
+              and activityType_d in (27, 33, 133, 134, 139, 3629)
+          | parse-kv DataFields_s as (ipAddress: string, username: string, userScope: string, accountName: string, fullScopeDetails: string, fullScopeDetailsPath: string, role: string, scopeLevel: string, source: string, sourceType: string) with (pair_delimiter=",", kv_delimiter=":", quote='"')
+          | lookup EventFieldsLookup on activityType_d
+          | lookup EventResultDetailsLookup on comments_s
+          | extend 
+              DvcIpAddr = iff(ipAddress == "null", "", ipAddress),
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml]: Done.
As discussed, this is the public IP of Network from where user tried to login, So we have considered it as SrcIpAddr now.

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+          | where event_name_s == "Alerts."
+              and alertInfo_eventType_s in ("WINLOGONATTEMPT", "WINLOGOFFATTEMPT")
+          | lookup EventTypeLookup on alertInfo_eventType_s
+          | lookup EventSubTypeLookup on alertInfo_loginType_s
+          | lookup DeviceTypeLookup on agentDetectionInfo_machineType_s
+          | invoke _ASIM_ResolveDvcFQDN('agentDetectionInfo_name_s')
+          | invoke _ASIM_ResolveDstFQDN('alertInfo_loginAccountDomain_s')
+          | extend
+              EventResult = iff(alertInfo_loginIsSuccessful_s == "true", "Success", "Failure"),
+              EventSeverity = iff(ruleInfo_severity_s == "Critical", "High", ruleInfo_severity_s)
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:120]: Yes

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+          | project-rename
+              EventStartTime = alertInfo_createdAt_t,
+              DvcIpAddr = alertInfo_srcMachineIp_s,
+              ActingAppName = sourceProcessInfo_name_s,
+              DvcId = agentDetectionInfo_uuid_g,
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:125]: As discussed, we have considered agents related field as Dvc field for mapping.

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+              TargetUserId = alertInfo_loginAccountSid_s,
+              TargetUsername = alertInfo_loginsUserName_s
+          | extend
+              Rule = RuleName,
+              ActingAppType = iff(isnotempty(ActingAppName), "Process", ""),
+              DvcIdType = iff(isnotempty(DvcId), "Other", "");
+      union activitydata, alertdata
+      | extend
+          EventCount = int(1),
+          EventProduct = "SentinelOne",
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:142]: Yes

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+              EventStartTime = createdAt_t,
+              TargetUserId = userId_s,
+              EventOriginalUid = activityUuid_g,
+              EventOriginalResultDetails = comments_s,
+              EventMessage = primaryDescription_s;
+      let alertdata = alldata
+          | where event_name_s == "Alerts."
+              and alertInfo_eventType_s in ("WINLOGONATTEMPT", "WINLOGOFFATTEMPT")
+          | lookup EventTypeLookup on alertInfo_eventType_s
+          | lookup EventSubTypeLookup on alertInfo_loginType_s
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:114]: No. As we are not getting data in Activities type of logs

File: Parsers/ASimAuthentication/Tests/SentinelOne_ASimAuthentication_SchemaTest.csv [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+"(2) Info: Missing optional field [DvcOriginalAction]"
+"(2) Info: Missing optional field [DvcScopeId]"
+"(2) Info: Missing optional field [DvcScope]"
+"(2) Info: Missing optional field [DvcZone]"
+"(2) Info: Missing optional field [EventOriginalSubType]"
+"(2) Info: Missing optional field [EventOwner]"
+"(2) Info: Missing optional field [EventProductVersion]"
+"(2) Info: Missing optional field [EventReportUrl]"
+"(2) Info: Missing optional field [HttpUserAgent]"
+"(2) Info: Missing optional field [LogonMethod]"
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Tests/SentinelOne_ASimAuthentication_SchemaTest.csv:25]: No.

File: Parsers/ASimAuthentication/Tests/SentinelOne_ASimAuthentication_SchemaTest.csv [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+"(2) Info: Missing optional field [LogonProtocol]"
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Tests/SentinelOne_ASimAuthentication_SchemaTest.csv:26]: No.

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+          | lookup DeviceTypeLookup on agentDetectionInfo_machineType_s
+          | invoke _ASIM_ResolveDvcFQDN('agentDetectionInfo_name_s')
+          | invoke _ASIM_ResolveDstFQDN('alertInfo_loginAccountDomain_s')
+          | extend
+              EventResult = iff(alertInfo_loginIsSuccessful_s == "true", "Success", "Failure"),
+              EventSeverity = iff(ruleInfo_severity_s == "Critical", "High", ruleInfo_severity_s)
+          | project-rename
+              EventStartTime = alertInfo_createdAt_t,
+              DvcIpAddr = alertInfo_srcMachineIp_s,
+              ActingAppName = sourceProcessInfo_name_s,
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:124]: No.

File: Parsers/ASimAuthentication/Tests/SentinelOne_vimAuthentication_DataTest.csv [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+"(2) Info: Empty value in 1246 records (98.89%)  in optional field [DvcOs] (Schema:Authentication)"
+"(2) Info: Empty value in 1246 records (98.89%)  in optional field [EventOriginalSeverity] (Schema:Authentication)"
+"(2) Info: Empty value in 1246 records (98.89%)  in optional field [EventSubType] (Schema:Authentication)"
+"(2) Info: Empty value in 1246 records (98.89%)  in optional field [RuleName] (Schema:Authentication)"
+"(2) Info: Empty value in 1246 records (98.89%)  in optional field [Rule] (Schema:Authentication)"
+"(2) Info: Empty value in 1246 records (98.89%)  in optional field [TargetDeviceType] (Schema:Authentication)"
+"(2) Info: Empty value in 1246 records (98.89%)  in recommended field [DvcHostname] (Schema:Authentication)"
+"(2) Info: Empty value in 1246 records (98.89%)  in recommended field [EventSeverity] (Schema:Authentication)"
+"(2) Info: Empty value in 1254 records (99.52%)  in optional field [TargetHostname] (Schema:Authentication)"
+"(2) Info: Empty value in 1254 records (99.52%)  in recommended field [Dst] (Schema:Authentication)"
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Tests/SentinelOne_vimAuthentication_DataTest.csv]: We don't have any fields related to Dst.

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 17]
```diff
@@ -0,0 +1,182 @@
...
+              EventOriginalResultDetails = comments_s,
+              EventMessage = primaryDescription_s;
+      let alertdata = alldata
+          | where event_name_s == "Alerts."
+              and alertInfo_eventType_s in ("WINLOGONATTEMPT", "WINLOGOFFATTEMPT")
+          | lookup EventTypeLookup on alertInfo_eventType_s
+          | lookup EventSubTypeLookup on alertInfo_loginType_s
+          | lookup DeviceTypeLookup on agentDetectionInfo_machineType_s
+          | invoke _ASIM_ResolveDvcFQDN('agentDetectionInfo_name_s')
+          | invoke _ASIM_ResolveDstFQDN('alertInfo_loginAccountDomain_s')
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml]: Updated. As alertInfo_loginAccountDomain_s contains Src info.

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 18]
```diff
@@ -0,0 +1,182 @@
...
+          DstFQDN,
+          Computer,
+          MG,
+          ManagementGroupName,
+          RawData,
+          SourceSystem,
+          TenantId,
+          _ResourceId
+  };
+  parser(disabled=disabled);
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml]: Done

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 19]
```diff
@@ -0,0 +1,182 @@
...
+  let EventResultDetailsLookup = datatable (comments_s: string, EventResultDetails: string)
+  [
+      "invalid 2FA code", "Incorrect password",
+      "IP/User mismatch", "No such user or password",
+      "invalid password", "Incorrect password",
+      "user temporarily locked 2FA attempt", "User locked",
+      "no active site", "Other"
+  ];
+  let EventFieldsLookup = datatable (
+      activityType_d: real,
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:35]: No. As activiType_d column of SentinelOne_CL table is of type real.

[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:35]: Done

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 20]
```diff
@@ -0,0 +1,182 @@
...
+          | lookup EventSubTypeLookup on alertInfo_loginType_s
+          | lookup DeviceTypeLookup on agentDetectionInfo_machineType_s
+          | invoke _ASIM_ResolveDvcFQDN('agentDetectionInfo_name_s')
+          | invoke _ASIM_ResolveDstFQDN('alertInfo_loginAccountDomain_s')
+          | extend
+              EventResult = iff(alertInfo_loginIsSuccessful_s == "true", "Success", "Failure"),
+              EventSeverity = iff(ruleInfo_severity_s == "Critical", "High", ruleInfo_severity_s)
+          | project-rename
+              EventStartTime = alertInfo_createdAt_t,
+              DvcIpAddr = alertInfo_srcMachineIp_s,
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml]: Done. As discussed, mapped alertInfo_srcMachineIp_s with SrcIpAddr

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml (hunk 2 above)
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:83]: As discussed we have mapped the role directly with TargetUserType when the value is similar to the allowed values, kept empty for role="null" and mapped to "Other" for other type of values.

File: Parsers/ASimAuthentication/Parsers/vimAuthenticationSentinelOne.yaml (hunk 3 above)
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/vimAuthenticationSentinelOne.yaml]: Done.

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml (hunk 4 above)
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml]: Done

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml (hunk 5 above)
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:53]: Done

File: Parsers/ASimAuthentication/Parsers/vimAuthenticationSentinelOne.yaml (hunk 6 above)
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/vimAuthenticationSentinelOne.yaml:1]: Done

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml (hunk 7 above)
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml]: Done

File: Parsers/ASimAuthentication/Tests/SentinelOne_ASimAuthentication_SchemaTest.csv [hunk 21]
```diff
@@ -0,0 +1,81 @@
...
+"(2) Info: Missing optional field [TargetOriginalRiskLevel]"
+"(2) Info: Missing optional field [TargetOriginalUserType]"
+"(2) Info: Missing optional field [TargetPortNumber]"
+"(2) Info: Missing optional field [TargetRiskLevel]"
+"(2) Info: Missing optional field [TargetSessionId]"
+"(2) Info: Missing optional field [TargetSessionId]"
+"(2) Info: Missing optional field [TargetSessionId]"
+"(2) Info: Missing optional field [TargetUrl]"
+"(2) Info: Missing optional field [TargetUserScopeId]"
+"(2) Info: Missing optional field [TargetUserType]"
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Tests/SentinelOne_ASimAuthentication_SchemaTest.csv]: Added TargetUserType field for Activities type of data based on role and for Alerts type of data by using function _ASIM_GetUserType(TargetUsername, TargetUserId).

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 22]
```diff
@@ -0,0 +1,182 @@
...
+                      "sourceType", sourceType
+                  )
+          | project-rename
+              EventStartTime = createdAt_t,
+              TargetUserId = userId_s,
+              EventOriginalUid = activityUuid_g,
+              EventOriginalResultDetails = comments_s,
+              EventMessage = primaryDescription_s;
+      let alertdata = alldata
+          | where event_name_s == "Alerts."
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:111]: We are not getting EventResultDetails for Alerts type of data.

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 23]
```diff
@@ -0,0 +1,182 @@
...
+          | invoke _ASIM_ResolveDstFQDN('alertInfo_loginAccountDomain_s')
+          | extend
+              EventResult = iff(alertInfo_loginIsSuccessful_s == "true", "Success", "Failure"),
+              EventSeverity = iff(ruleInfo_severity_s == "Critical", "High", ruleInfo_severity_s)
+          | project-rename
+              EventStartTime = alertInfo_createdAt_t,
+              DvcIpAddr = alertInfo_srcMachineIp_s,
+              ActingAppName = sourceProcessInfo_name_s,
+              DvcId = agentDetectionInfo_uuid_g,
+              DvcOs = agentDetectionInfo_osName_s,
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:126]: As discussed, we have considered agents-related field as the Dvc field for mapping.

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 24]
```diff
@@ -0,0 +1,182 @@
...
+      EventType: string,
+      EventResult: string,
+      EventOriginalType: string
+  )
+  [
+      27, "Logon", "Success", "User Logged In",
+      33, "Logoff", "Success", "User Logged Out",
+      133, "Logon", "Failure", "Existing User Login Failure",
+      134, "Logon", "Failure", "Unknown User Login",
+      139, "Logon", "Failure", "User Failed to Start an Unrestricted Session",
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:45]: As discussed, We have kept EventType=Logon only.

**Comment by v-sudkharat** [comment]
Hi @jayeshprajapaticrest, thank you for accommodating the suggested changes! We will review the Pull Request and get back to you by shortly.  Thanks!

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 25]
```diff
@@ -0,0 +1,185 @@
+Parser:
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:1]: @jayeshprajapaticrest did you find any alert logs from SentinelOne any schema related?

Logs which can be mapped to these fields with https://learn.microsoft.com/en-us/azure/sentinel/normalization-schema-authentication#inspection-fields

We have these fields in each schema

**Comment by v-sudkharat** [comment]
Hello @jayeshprajapaticrest, please check above suggested changes. Thanks

**Comment by keyur-cds** [comment]
@v-sudkharat We will look into these comments and will update here once those are worked upon.

Thanks!

File: Parsers/ASimAuthentication/Tests/SentinelOne_ASimAuthentication_DataTest.csv [hunk 26]
```diff
@@ -0,0 +1,36 @@
+﻿Result
+"(0) Error: 1 invalid value(s) (up to 10 listed) in 1263 records (100.0%) for field [EventProduct] of type [Enumerated]: [""SentinelOne""] (Schema:Authentication)"
+"(0) Error: 1 invalid value(s) (up to 10 listed) in 1263 records (100.0%) for field [EventVendor] of type [Enumerated]: [""SentinelOne""] (Schema:Authentication)"
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Tests/SentinelOne_ASimAuthentication_DataTest.csv:3]: pls update this in ASIM tester

**Review by vakohl — CHANGES_REQUESTED** [review]

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml [hunk 27]
```diff
@@ -0,0 +1,185 @@
...
+  Title: ASIM Authentication parser for SentinelOne
+  Version: '0.1.0'
+  LastUpdated: Jul 24 2023
```
[the remark this answers is not in the document]
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml]: Done

**Comment by niralishah-crest** [comment]
@vakohl we have mapped the inspection fields below in the Authentication parser, which we found from SentinelOne Alerts logs.
RuleName
Rule
ThreatConfidence
ThreatOriginalConfidence

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml (hunk 25 above)
Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:1]: @jayeshprajapaticrest 
No other Threat field info coming in SentinelOne Authentication?
The logs I had only have 'threatconfidence' as '15', hope you have validated with various sample values i.e. 75, 80, 90, 100 etc.

File: Parsers/ASimAuthentication/Tests/SentinelOne_ASimAuthentication_DataTest.csv (hunk 26 above)
Comment (vakohl) [reply Parsers/ASimAuthentication/Tests/SentinelOne_ASimAuthentication_DataTest.csv:3]: checked

File: Parsers/ASimAuthentication/Parsers/ASimAuthentication.yaml [hunk 28]
```diff
@@ -61,4 +62,5 @@ Parsers:
  - ASimAuthenticationSshd
  - ASimAuthenticationSu
  - ASimAuthenticationVectraXDRAudit
+ - ASimAuthenticationSentinelOne
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthentication.yaml:65]: the format is not correct in Authentication, pls use below with underscore:
image

File: Parsers/ASimAuthentication/Parsers/imAuthentication.yaml [hunk 29]
```diff
@@ -67,4 +68,5 @@ Parsers:
   - vimAuthenticationCiscoISE
   - vimAuthenticationBarracudaWAF
   - vimAuthenticationVectraXDRAudit
+  - vimAuthenticationSentinelOne
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/imAuthentication.yaml:71]: pls use below format:
image

File: Parsers/ASimAuthentication/Parsers/ASimAuthentication.yaml (hunk 28 above)
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/ASimAuthentication.yaml]: Done

File: Parsers/ASimAuthentication/Parsers/imAuthentication.yaml (hunk 29 above)
Comment (jayeshprajapaticrest) [reply Parsers/ASimAuthentication/Parsers/imAuthentication.yaml]: Done

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml (hunk 25 above)
Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml:1]: Author commented on other PR that they didn't had actual data but have tested it by ingesting curated test data

**Review by vakohl — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**